pkg/controller/privatelink/actuator/awsactuator/awshubactuator.go

```diff
@@ -567,27 +576,71 @@ func (a *AWSHubActuator) getAssociatedVPCs(
 		cd.Status.Platform.AWS.PrivateLink != nil &&
 		cd.Status.Platform.AWS.PrivateLink.VPCEndpointID != "" {
 
-		endpointResp, err := a.awsClientHub.DescribeVpcEndpoints(&ec2.DescribeVpcEndpointsInput{
-			Filters: []*ec2.Filter{ec2FilterForCluster(metadata)},
-		})
+		endpointVPC, err := a.getEndpointVPC(cd, metadata)
 		if err != nil {
-			return associatedVPCs, errors.Wrap(err, "error getting the VPC Endpoint")
+			return associatedVPCs, err
 		}
 
-		if len(endpointResp.VpcEndpoints) == 0 {
-			return associatedVPCs, nil // no vpc endpoint
+		if endpointVPC.VPCID != "" {
+			logger.Debugf("adding VpcEndpoint VPC %s", endpointVPC.VPCID)
+			associatedVPCs = append(associatedVPCs, endpointVPC)
 		}
+	}
 
-		logger.Debugf("adding VpcEndpoint [%s] VPC [%s]", *endpointResp.VpcEndpoints[0].VpcEndpointId, *endpointResp.VpcEndpoints[0].VpcId)
-		endpointVPC := hivev1.AWSAssociatedVPC{
-			AWSPrivateLinkVPC: hivev1.AWSPrivateLinkVPC{
-				VPCID:  *endpointResp.VpcEndpoints[0].VpcId,
-				Region: cd.Spec.Platform.AWS.Region,
-			},
+	return associatedVPCs, nil
+}
+
+func (a *AWSHubActuator) getEndpointVPC(cd *hivev1.ClusterDeployment, metadata *hivev1.ClusterMetadata) (hivev1.AWSAssociatedVPC, error) {
+	endpointVPC := hivev1.AWSAssociatedVPC{}
+	endpointResp, err := a.awsClientHub.DescribeVpcEndpoints(&ec2.DescribeVpcEndpointsInput{
+		Filters: []*ec2.Filter{ec2FilterForCluster(metadata)},
+	})
+	if err != nil {
+		return endpointVPC, errors.Wrap(err, "error getting the VPC Endpoint")
+	}
+
+	if len(endpointResp.VpcEndpoints) >= 0 {
+		endpointVPC.VPCID = *endpointResp.VpcEndpoints[0].VpcId
+		endpointVPC.Region = cd.Spec.Platform.AWS.Region
+	}
+
+	return endpointVPC, nil
+}
+
+func (a *AWSHubActuator) selectHostedZoneVPC(cd *hivev1.ClusterDeployment, metadata *hivev1.ClusterMetadata, logger log.FieldLogger) (hivev1.AWSAssociatedVPC, error) {
+	selectedVPC := hivev1.AWSAssociatedVPC{}
+
+	// For clusterdeployments that are on AWS, use the VPCEndpoint VPC
+	if cd.Status.Platform != nil &&
+		cd.Status.Platform.AWS != nil &&
+		cd.Status.Platform.AWS.PrivateLink != nil &&
+		cd.Status.Platform.AWS.PrivateLink.VPCEndpointID != "" {
+
+		endpointVPC, err := a.getEndpointVPC(cd, metadata)
+		if err != nil {
+			return selectedVPC, errors.Wrap(err, "error getting Endpoint VPC")
 		}
 
-		associatedVPCs = append(associatedVPCs, endpointVPC)
+		if endpointVPC.VPCID == "" {
+			return selectedVPC, errors.New("unable to select Endpoint VPC: Endpoint not found")
+		}
+
+		return endpointVPC, nil
 	}
```

Comment:
I feel like there's an opportunity to DRY some of this -- though I can see why you didn't.

Perhaps `getEndpointVPC()`'s return could indicate whether `VPCEndpointID` was unset in the status?